include/beast/websocket/stream.hpp

```diff
@@ -627,6 +627,14 @@ class stream
         return wr_opcode_ == detail::opcode::text;
     }
 
+
+    /// Returns `true` if either the server or the client sent a close frame.
+    bool 
+    closed() const
+    {
+        return wr_close_ || rd_close_;
```

Comment:
Now that I look at it more closely, there's a smell coming out of this close code...you are on to something!